game_of_life_interface.py

```diff
@@ -0,0 +1,44 @@
+"""Conway's Game of Life interface.
+
+This module contains the user inferface for the game as well as
+the rules for how the next generation is created. The rules of the 
+game are as follows:
+
+    1) Any live cell with two or three live neighbors survives.
+    2) Any dead cell with three live neighbors becomes a live cell.
+    3) All other live cells die in the next generation. Similarly, 
+    all other dead cells stay dead.
+
+After each generation the board is updated for the user to see. 
+
+TODO: Implement as web app with Javascript (React?) frontend.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+import game_of_life
+
+
+print("Let the Game of life begin!")
+
+# TODO: allow user to select the game board size
+game = game_of_life.Board((12, 12)) # Initialize with randomly seeded board
+plt.ion() # set matplotlib to interactive mode
+img = plt.imshow(game.board) # Show the game board as image
+# TODO: allow user to select number of generations to play
+for n in range(100): # Set number of generations / life cycles to play
+    next_generation = game.board.copy()
```

Comment:
Is it possible you want `deepcopy` here? I don't know much about numpy arrays, but if you were using a 2-D array (i.e. `self.board = [[random.randint(0,1) for c in num_cols] for r in num_rows]`, then `copy` would create a new list full of references to the original's inner lists which probably isn't what you would want.

Reply:
Thanks, for pointing this out! Numpy syntax for deep and shallow copies is different from standard Python. In Numpy, view() is used for shallow copies and copy() is used for deep copies. Kind of confusing when Python uses copy() for shallow copying and deepcopy() for deep copies.

Reply:
@lbernick , thanks for pointing this out! Numpy syntax for deep and shallow copies is different from standard Python. In Numpy, view() is used for shallow copies and copy() is used for deep copies. Kind of confusing when Python uses copy() for shallow copying and deepcopy() for deep copies.

Reply:
Interesting, thanks! TIL